tnt/src/main/java/org/dcsa/conformance/standards/tnt/party/TntPublisher.java

```diff
@@ -112,13 +130,167 @@ private void supplyScenarioParameters(JsonNode actionPrompt) {
+      List<SortCriteria> sortCriteria = sortingFields.stream()
+        .map(sortField -> {
+          String[] parts = sortField.split(":");
+          return new SortCriteria(parts[0], attributeMappings.get(parts[0]),
+            parts.length > 1 ? SortDirection.valueOf(parts[1].toUpperCase()) : SortDirection.ASCENDING);
+        })
+        .collect(Collectors.toList());
+      filteredArray = sortJsonArray(filteredArray, sortCriteria);
+
+    }
+    // Sort the filtered array by user provided value in parameters.
+    ArrayNode limitedArray = OBJECT_MAPPER.createArrayNode();
+    if (request.queryParams().containsKey("limit"))  {
+      int limit = Integer.parseInt(request.queryParams().get("limit").iterator().next());
+      if (filteredArray.size() > limit) {
+        StreamSupport.stream(filteredArray.spliterator(), false)
+          .limit(limit)
+          .forEach(limitedArray::add);
+      }
+    }
     return request.createResponse(
         200,
         Map.of(API_VERSION, List.of(apiVersion)),
-        new ConformanceMessageBody(jsonResponseBody));
+        new ConformanceMessageBody(limitedArray.isEmpty() ? filteredArray : limitedArray ));
+  }
+
+  public static ArrayNode sortJsonArray(ArrayNode jsonArray, List<SortCriteria> criteria) {
+    if (jsonArray == null || criteria == null || criteria.isEmpty()) {
+      return jsonArray; // No sorting needed
+    }
+
+    List<JsonNode> jsonNodeList = new ArrayList<>();
+    jsonArray.forEach(jsonNodeList::add);
+
+    jsonNodeList.sort((node1, node2) -> {
+      int comparisonResult = 0;
+      for (SortCriteria criterion : criteria) {
+        comparisonResult = compareNodesByField(node1, node2, criterion.getField(), criterion.getDirection());
+        if (comparisonResult != 0) {
+          break; // Stop comparing if a difference is found
+        }
+      }
+      return comparisonResult;
+    });
+
+    ArrayNode sortedArray = jsonArray.arrayNode();
+    jsonNodeList.forEach(sortedArray::add);
+    return sortedArray;
+  }
+
+  private ConformanceResponse return400(ConformanceRequest request, String message) {
+    return request.createResponse(
+      400,
+      Map.of(API_VERSION, List.of(apiVersion)),
+      new ConformanceMessageBody(createErrorResponse(request.method(), request.url(), "Bad Request", message)));
+  }
+
+  private ConformanceResponse return404(ConformanceRequest request) {
+    return return404(request, "Returning 404 since the request did not match any known URL");
   }
+
+  private ConformanceResponse return404(ConformanceRequest request, String message) {
+    return request.createResponse(
+      404,
+      Map.of(API_VERSION, List.of(apiVersion)),
+      new ConformanceMessageBody(createErrorResponse(request.method(), request.url(), "Not Found", message)));
+  }
+
+  private ObjectNode createErrorResponse(String httpMethod, String requestUri, String reason, String message) {
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
Please remove this new mapper, and use `OBJECT_MAPPER`.

Reply:
done